src/Storage/Controllers/InstancesController.cs

```diff
@@ -141,51 +141,53 @@ public class InstancesController : ControllerBase
                 return BadRequest("Both InstanceOwner.PartyId and InstanceOwnerIdentifier cannot be present at the same time.");
             }
 
-            if (orgClaim != null)
+            switch ((orgClaim != null, userId != null))
             {
-                if (!_authorizationService.UserHasRequiredScope(_generalSettings.InstanceReadScope))
-                {
-                    return Forbid();
-                }
-
-                if (string.IsNullOrEmpty(org) && string.IsNullOrEmpty(appId))
-                {
-                    return BadRequest("Org or AppId must be defined.");
-                }
-
-                org = string.IsNullOrEmpty(org) ? appId.Split('/')[0] : org;
-
-                if (!orgClaim.Equals(org, StringComparison.InvariantCultureIgnoreCase))
-                {
-                    return Forbid();
-                }
+                case (true, _):
+                    if (!_authorizationService.UserHasRequiredScope(_generalSettings.InstanceReadScope))
+                    {
+                        return Forbid();
+                    }
 
-                appOwnerRequestingInstances = true;
-            }
-            else if (userId != null)
-            {
-                if (instanceOwnerPartyId == null)
-                {
-                    if (string.IsNullOrEmpty(instanceOwnerIdentifier))
+                    if (string.IsNullOrEmpty(org) && string.IsNullOrEmpty(appId))
                     {
-                        return BadRequest("Either InstanceOwnerPartyId or InstanceOwnerIdentifier need to be defined.");
+                        return BadRequest("Org or AppId must be defined.");
                     }
 
-                    (string instanceOwnerIdType, string instanceOwnerIdValue) = InstanceHelper.GetIdentifierFromInstanceOwnerIdentifier(instanceOwnerIdentifier);
+                    org = string.IsNullOrEmpty(org) ? appId.Split('/')[0] : org;
 
-                    if (string.IsNullOrEmpty(instanceOwnerIdType) || string.IsNullOrEmpty(instanceOwnerIdValue))
+                    if (!orgClaim.Equals(org, StringComparison.InvariantCultureIgnoreCase))
                     {
-                        return BadRequest("Invalid InstanceOwnerIdentifier.");
+                        return Forbid();
                     }
 
-                    (string person, string orgNo) = InstanceHelper.SeparatePersonAndOrgNo(instanceOwnerIdType, instanceOwnerIdValue);
+                    appOwnerRequestingInstances = true;
+                    break;
 
-                    instanceOwnerPartyId = await _registerService.PartyLookup(person, orgNo);
-                }
+                case (false, true):
+                    break;
+
+                default:
+                    return BadRequest();
             }
-            else
+
+            if (instanceOwnerPartyId == null)
```

Comment:
Instance owner party id is required only for end users. If the caller is org, then instance owner is optional.